contrib/src/templates/context.rs

```diff
@@ -11,20 +13,41 @@ pub struct Context {
     pub root: PathBuf,
     /// Mapping from template name to its information.
     pub templates: HashMap<String, TemplateInfo>,
-    /// Mapping from template name to its information.
-    pub engines: Engines
+    /// Loaded template engines, or None if there was a failure during a reload
+    pub engines: Option<Engines>,
```

Comment:
Why is this an `Option`? If there was a failure, we should keep using the old engines.